### PR TITLE
test: add shuffle circuit test

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR adds shuffle circuit test to test multi phase.

### CODE REVIEW

File: vendors/halo2/src/circuits/simple_circuit.rs [hunk 1]
```diff
@@ -585,7 +585,7 @@ mod test {
             tachyon_create_proof::<_, _, _, _, _>(
                 &mut prover,
                 &mut tachyon_pk,
-                &[circuit.clone()],
+                &[circuit.clone(), circuit.clone()],
```
Comment (TomTaehoonKim) [inline vendors/halo2/src/circuits/simple_circuit.rs:590]:
```suggestion
                &[circuit.clone(), circuit],
```

File: vendors/halo2/src/bn254_gwc_prover.cc [hunk 2]
```diff
@@ -98,7 +98,12 @@ void GWCProver::create_proof(ProvingKey& key,
 
   tachyon_halo2_bn254_argument_data* data =
       tachyon_halo2_bn254_argument_data_create(num_circuits);
+
```
Comment (TomTaehoonKim) [inline vendors/halo2/src/bn254_gwc_prover.cc:101]: Is there a particular reason you added a empty line here?

File: vendors/halo2/src/bn254_shplonk_prover.cc [hunk 3]
```diff
@@ -101,7 +101,12 @@ void SHPlonkProver::create_proof(ProvingKey& key,
 
   tachyon_halo2_bn254_argument_data* data =
       tachyon_halo2_bn254_argument_data_create(num_circuits);
+
```
Comment (TomTaehoonKim) [inline vendors/halo2/src/bn254_shplonk_prover.cc:104]: Is there a particular reason you added a empty line here?

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

File: vendors/halo2/src/bn254_gwc_prover.cc (hunk 2 above)
Comment (chokobole) [reply vendors/halo2/src/bn254_gwc_prover.cc:101]: To put `reserve` and `add` together?

File: vendors/halo2/src/bn254_shplonk_prover.cc (hunk 3 above)
Comment (chokobole) [reply vendors/halo2/src/bn254_shplonk_prover.cc:104]: To put `reserve` and `add` together?

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Pull request merged**